optimum/amd/ryzenai/utils.py

```diff
@@ -21,3 +40,90 @@ def validate_provider_availability(provider: str):
         raise ValueError(
             f"Asked to use {provider} as an ONNX Runtime execution provider, but the available execution providers are {available_providers}."
         )
+
+
+def set_builtins():
+    """Set the builtins.impl and builtins.quant_mode environment variables."""
+    builtins.impl = os.getenv("BUILTINS_IMPL", DEFAULT_BUILTIN_IMPL)
+    builtins.quant_mode = os.getenv("BUILTINS_IMPL", DEFAULT_BUILTIN_QUANT_MODE)
+    print(f"Builtins: impl={builtins.impl}, quant_mode={builtins.quant_mode}")
+
+
+def clone_repository(repo_url: str, repo_path: str):
+    try:
+        if not os.path.exists(repo_path):
+            subprocess.run(["git", "clone", "--depth", "1", "--branch", "main", repo_url, repo_path], check=True)
+            subprocess.run(["git", "-C", repo_path, "checkout", RYZEN_SW_COMMIT_HASH], check=True)
+    except subprocess.CalledProcessError as e:
+        print(f"Error: {e}")
+        if os.path.exists(repo_path):
+            shutil.rmtree(repo_path)
+
+
+def set_env_var(key, value):
+    if key not in os.environ:
+        os.environ[key] = value
+
+
+def normalize_path(path):
+    return os.path.normpath(path)
+
+
+def check_env_path_exists(env_var_name):
+    paths = os.environ.get(env_var_name)
+    if ";" in paths:
+        paths = paths.strip(";").split(";")
+    elif "," in paths:
+        paths = paths.strip(",").split(",")
+    else:
+        paths = [paths]
+
+    for path in paths:
+        if not os.path.exists(path):
+            raise OSError(
+                f"The path '{path}' does not exist. Please ensure that the `{env_var_name}` environment variable is set correctly!"
+            )
+
+
+def set_environment_variables():
+    ryzenai_sw_path = os.environ.get("RYZENAI_SW_PATH")
+    if not ryzenai_sw_path:
+        logger.warning(
+            "RYZENAI_SW_PATH environment variable is not set. Attempting to clone RyzenAI-SW repository now...\n"
+        )
+        ryzenai_sw_path = normalize_path(os.path.join(os.getcwd(), "RyzenAI-SW"))
+        clone_repository("https://github.com/amd/RyzenAI-SW/", ryzenai_sw_path)
```

Comment:
Couldn't we instead have https://github.com/amd/RyzenAI-SW as a git submodule with a pinned commit (in order to avoid breaking changes)?

No strong opinion here, could anyway revisit later. Still probably a good idea to pin the commit.

Reply:
I am using a specific commit after clone to avoid breaking changes. Added the `RYZEN_SW_COMMIT_HASH` at top of file.